Hermes/handlers.py

```diff
@@ -0,0 +1,172 @@
+""" Handlers for accepting HTTP requests. """
+
+import json
+import logging
+import Queue
+import threading
+from tornado.ioloop import IOLoop
+from tornado.web import RequestHandler
+
+import hermes_constants
+import helper
+from helper import JSONTags
+from helper import NodeInfoTags
+from helper import TASK_STATUS
+from helper import TASK_STATUS_LOCK
+
+# A list of required parameters that define a task.
+REQUIRED_KEYS = ['task_id', 'type', 'bucket_name']
+
+class TaskStatus(object):
+  """ A class containing all possible task states. """
+  PENDING = 'pending'
+  FAILED = 'failed'
+  SUCCESSFUL = 'successful'
+
+class MainHandler(RequestHandler):
+  """ Main handler class. """
+
+  # The path for this handler.
+  PATH = "/"
+
+  def get(self):
+    """ Main GET method. Reports the status of the server. """
+    self.write(json.dumps({'status': 'up'}))
+
+class PollHandler(RequestHandler):
+  """ Handler that polls for new tasks. """
+
+  # The path for this handler.
+  PATH = "/poll"
+
+  def get(self):
+    """ GET method that polls for a new task. """
+
+    # Send request to AppScale Portal.
+    logging.info("Sending request to AppScale Portal.")
+    url = "{0}{1}".format(hermes_constants.PORTAL_URL,
+        hermes_constants.PORTAL_POLL_PATH)
+    data = json.dumps({
+      JSONTags.DEPLOYMENT_ID: helper.get_deployment_id()
+    })
+    request = helper.create_request(url=url, method='POST', body=data)
+    response = helper.urlfetch(request)
+    if not response[JSONTags.DEPLOYMENT_ID]:
+      self.set_status(hermes_constants.HTTP_Codes.HTTP_OK)
```

Comment:
I haven't seen what the portal is supposed to return, but should this be logged?